File: stackBarebone/Stack.c
#include <stdio.h>

typedef int T;
/* const int MAX STACK = 5; */
#define MAX_STACK 5
T stack[MAX_STACK];
int top = 0;

T item = 1234;

void print(){
	for(int i=0; i < MAX_STACK; i++) printf("%d ",stack[i]);
}

int main(){
	stack[top++] = item; // push
	print();
	item = stack[--top]; // pop
	
	//top = 5;
	//stack[0]=1; 
	//--top;

	return 0;
}

File: JavaOOP/stackJava/Stack.java
package stackJava;


class Stack<T> {
	
    private int top, size;
	private T[] stack;
    private String type;

	public Stack (int size) {
        this.top = 0; 
        this.size = size; 
        this.stack= (T[]) new Object[size];
        //int [] i = new int[size]; / array id declaration and allocation
        //int x[10]; //error
    }
	
    public void push (T item) throws StackOverflowException{
        if (this.is_full()) throw new StackOverflowException(this); 
        this.stack[this.top++] = item;
    }
	public T pop ()throws StackUnderflowException{
        if (this.is_empty()) throw new StackUnderflowException(this); 
        return 	this.stack[--this.top];

    }
	public boolean is_empty (){ return this.top == 0;}
	public boolean is_full (){ return this.top == this.size;}
		//…
       
    void setType(String type){this.type = type;}
    String getType(){return this.type;}
	
};
